tree/treeplayer/src/TTreeIndex.cxx

```diff
@@ -430,8 +433,8 @@ Long64_t TTreeIndex::GetEntryNumberWithBestIndex(Long64_t major, Long64_t minor)
 /// Return entry number corresponding to major and minor number.
 /// Note that this function returns only the entry number, not the data
 /// To read the data corresponding to an entry number, use TTree::GetEntryWithIndex
-/// the BuildIndex function has created a table of Double_t* of sorted values
-/// corresponding to val = major<<31 + minor;
+/// the BuildIndex function has created two tables of Long64_t sorted values
+/// (with an internal intermediate cast to LongDouble)
 /// The function performs binary search in this sorted table.
 /// If it finds a pair that maches val, it returns directly the
 /// index in the table, otherwise it returns -1.
```

Comment:
```suggestion
/// index in the table, otherwise it returns -1.
/// \warning Due to internal architecture details, the maximum value for `(major, minor)`
/// for which the function works correctly and consistently in all platforms is `0xFFFFFFFFFFFF`, which is less than `kMaxLong64`.
```